craft_application/commands/lifecycle.py

```diff
@@ -269,6 +269,7 @@ def _run(
         """Run the prime command."""
         super()._run(parsed_args, step_name=step_name)
 
+        self._services.package.update_project()
```

Comment:
This could return the updated project to be set in the `write_metadata` call below and make the project class immutable, but it would change the existing method signature.